### PR TITLE
[WIP]Test caasp kvm

### PR DESCRIPTION
**Pull request by grahamhayes**

### CODE REVIEW

File: vars/cleanupEnvironment.groovy [hunk 1]
```diff
@@ -15,34 +15,8 @@ def call(Map parameters = [:]) {
...
-                    "WGET_FLAGS=--no-verbose",
-                    "NO_COLOR=true",
-                ]) {
-                    ansiColor('xterm') {
-                        sh(script: 'set -o pipefail; ./contrib/libvirt/k8s-libvirt.sh destroy 2>&1 | tee ${WORKSPACE}/logs/terraform-destroy.log')
-                    }
-                }
-            }
+        dir("automation/caasp-kvm") {
+            sh(script: "terraform apply -no-color -var 'kubic_salt_dir=${WORKSPACE}/salt' -var 'kubic_caasp_container_manifests_dir=${WORKSPACE}/caasp_container_manifests' -var 'caasp_container_manifests=channel://devel' 2>&1 | tee ${WORKSPACE}/logs/terraform-destroy.log")
```
Comment (kiall) [inline vars/cleanupEnvironment.groovy]: `set -o pipefail ;` is needed, if this fails to cleanup, it will leave a dirty slave hanging around.

Comment (grahamhayes) [inline vars/cleanupEnvironment.groovy]: It is also doing a `terraform apply` instead of destroying

**Comment by MaximilianMeister** [comment]
👍 @grahamhayes you can rebase the commit from https://github.com/kubic-project/jenkins-library/pull/27 and just drop the `DEVENV=true`

**Comment by MaximilianMeister** [comment]
@grahamhayes https://github.com/kubic-project/jenkins-library/pull/27 got merged, you might need to rebase again

File: vars/createEnvironment.groovy [hunk 2]
```diff
@@ -21,58 +21,33 @@ Environment call(Map parameters = [:]) {
...
-                    minion.ipv4 = tfMinion.ipv4
-                    minion.minionId = tfMinion.minionId != null ? tfMinion.minionId : tfMinion.minionID
-
-                    environment.minions.push(minion)
-                }
-
-                sh(script: "cp environment.json ${WORKSPACE}/logs/environment.json")
-                sh(script: "cat ${WORKSPACE}/logs/environment.json")
+            timeout(15){
+                sh(script: "./tools/wait_for_velum.py https://${environment.dashboardHost}")
```
Comment (MaximilianMeister) [inline vars/createEnvironment.groovy]:
```
Failure/Error: visit "/users/sign_up"
```
might be related to this here. could you try to query the `/users/sign_up` endpoint? I think this is still the stage where velum responds, but the database is not yet up and running. ...although i might be wrong as I couldn't find the code for `wait_for_velum.py`

File: vars/bootstrapEnvironment.groovy [hunk 3]
```diff
@@ -28,9 +28,8 @@ Environment call(Map parameters = [:]) {
         dir('automation/velum-bootstrap') {
             withEnv([
                 "VERBOSE=true",
-                "ENVIRONMENT=${WORKSPACE}/terraform/environment.json",
-                // TODO: drop this after switching to a VM based admin setup
-                "DEVENV=true",
+                "ENVIRONMENT=${WORKSPACE}/automation/caasp-kvm/environment.json",
+                "DEVENV=false",
```
Comment (MaximilianMeister) [inline vars/bootstrapEnvironment.groovy]: I have fixed this in https://github.com/kubic-project/automation/pull/12 so we don't need to specify `DEVENV=false` anymore

**Comment by MaximilianMeister** [comment]
@grahamhayes @kiall wouldn't it make more sense to not again use libvirt on openstack, but instead integrate it more into openstack itself?

@mjura has started already to work on that part https://github.com/kubic-project/automation/pull/30

this setup is much better than the devenv i agree, but it's still not ready, and i am wondering if it's not better to invest time in a pure openstack integration for the CI. I believe it will also be much more performant, as we wont have nested virtualization anymore

**Pull request closed without merging**